cmd/integration_link.go

```diff
@@ -0,0 +1,429 @@
+			if len(c.Args()) != 0 {
+				cli.ShowCommandHelp(c, "integration-link")
+				return
+			}
+
+			currentApp := appdetect.CurrentApp(c)
+			err := integrationlink.Show(currentApp)
+			if err != nil {
+				errorQuit(err)
+			}
+		},
+		BashComplete: func(c *cli.Context) {
+			_ = autocomplete.CmdFlagsAutoComplete(c, "integration-link")
+		},
+	}
+
+	integrationLinkCreateCommand = cli.Command{
+		Name:     "integration-link-create",
+		Category: "Integration Link",
+		Flags: []cli.Flag{
+			appFlag,
+			cli.StringFlag{Name: "branch", Usage: "Branch used in auto deploy"},
+			cli.BoolFlag{Name: "auto-deploy", Usage: "Enable auto deploy of application after each branch change"},
+			cli.BoolFlag{Name: "deploy-review-apps", Usage: "Enable auto deploy of review app when new pull request is opened"},
+			cli.BoolFlag{Name: "destroy-on-close", Usage: "Auto destroy review apps when pull request is closed"},
+			cli.BoolFlag{Name: "no-auto-deploy", Usage: "Enable auto deploy of application after each branch change"},
+			cli.BoolFlag{Name: "no-deploy-review-apps", Usage: "Enable auto deploy of review app when new pull request is opened"},
+			cli.BoolFlag{Name: "no-destroy-on-close", Usage: "Auto destroy review apps when pull request is closed"},
+			cli.UintFlag{Name: "hours-before-destroy-on-close", Usage: "Time delay before auto destroying a review app when pull request is closed"},
+			cli.BoolFlag{Name: "destroy-on-stale", Usage: "Auto destroy review apps when no deploy/commits has happened"},
+			cli.BoolFlag{Name: "no-destroy-on-stale", Usage: "Auto destroy review apps when no deploy/commits has happened"},
+			cli.UintFlag{Name: "hours-before-destroy-on-stale", Usage: "Time delay before auto destroying a review app when no deploy/commits has happened"},
+		},
+		Usage: "Link your Scalingo application to an integration",
+		Description: ` Link your Scalingo application to an integration:
+	$ scalingo --app my-app integration-link-create <repository URL> [options]
+									   OR
+	$ scalingo --app my-app integration-link-create <repository URL> [options]
+
+	List of available integrations:
+	- github => GitHub.com
+	- github-enterprise => GitHub Enterprise (private instance)
+	- gitlab => GitLab.com
+	- gitlab-self-hosted => GitLab Self-hosted (private instance)
+
+	Examples:
+	$ scalingo --app my-app integration-link-create https://gitlab.com/gitlab-org/gitlab-ce
+	$ scalingo --app my-app integration-link-create https://ghe.example.org/test/frontend-app --branch master --auto-deploy
+
+		# See also 'integration-link', 'integration-link-update', 'integration-link-delete', 'integration-link-manual-deploy' and 'integration-link-manual-review-app'`,
+		Action: func(c *cli.Context) {
+			if len(c.Args()) != 1 {
+				cli.ShowCommandHelp(c, "integration-link-create")
+				return
+			}
+
+			currentApp := appdetect.CurrentApp(c)
+			integrationURL := c.Args()[0]
+			integrationType, err := scmintegrations.GetTypeFromURL(integrationURL)
```

Comment:
Could you move all the "logic" to the scm_integrations package? In this package all Actions are quite small and just delegate the logic to the specific package. I find this easier to read.